src/services/fileStorage/strategies/awsS3.js

```diff
@@ -478,7 +478,7 @@ class AWSS3Strategy extends AbstractFileStorageStrategy {
 
 				return createAWSObject(result.schoolId).then((awsObject) => {
 					const params = {
-						Bucket: awsObject.bucket,
+						Bucket: bucket || awsObject.bucket,
```

Comment:
I don't quite get the change with the bucket here. Can you explain?

Reply:
@dyedwiper sure, before the bucket name used for the accessed object (for generatin the pre-signed GetObject operation URL) used to be the one generated as "bucket-" + school ID which caused the issue of files not being accessible after you've, e.g., changed the creator of a file (and within the KNL we've added the code that deletes this reference). And with this change we're able to pass the bucket name from the file object from the database (which we have saved within the document already, but it's just not used properly right now) to this function which generates the pre-signed URL for us. So now with this implementation it'll be possible to pass the bucket name or just go the old route in the cases it's still needed this way.

The issue is also well described within this ticket: https://ticketsystem.dbildungscloud.de/browse/BC-2469 . I think it's exactly the same issue, but we'll retest this one in the next sprint to make sure (and make it as resolved if that's the case).

Reply:
Okay, I can see that it might fix the problem for users from different schools, but for deleted users this line should never be reached because of `if (!result || !result.schoolId) { return new NotFound('User not found'); }` in line 475. Or am I missing something?

Reply:
It's an ID of a user trying to access the file (or rather generate the pre-signed S3's GetObject operation URL) so it should always be an ID of some user present within the system - it doesn't matter if the creator of the file is still there or not.

Reply:
Oh yeah, I see it in the call in the proxy-service. Thanks for pointing it out!